### ISSUE TITLE
Issues on server - no admin user created

### ISSUE THREAD

**Issue by Kinzi** [mrvautin/openKB#142]
I'm trying to install openKB on my server in a subfolder (local was fine) and have a few issues:

- somehow no admin user is created and I don't have access to settings. (I tried deleting all the db files to start fresh and I do fill out the form on /setup)
- also after login I don't get redirected, page stays on /login
- logo.png is a 404

Any idea what's going on there?

Permissions should be fine (775 - same user as my other node apps on this server)
nginx looks like this (again, same setup as other node apps):

```
server {
        listen 443 ssl;

 ...
...

location /docs {
                proxy_pass http://127.0.0.1:4444;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection 'upgrade';
                proxy_set_header Host $host;
                proxy_cache_bypass $http_upgrade;
        }
}

```

**Comment by mrvautin** [comment]
If you are wanting to run openKB with a URL something like: `http://domain.com/myopenkb`, you will have you set the `app_context` value to `myopenkb`.

**Comment by Kinzi** [comment]
@mrvautin  Yeah, I did that. Sorry forgot to mention. If you want you can take alook here: https://mydash.io/docs/

**Comment by Kinzi** [comment]
Is the first/admin user stored anywhere else but in `users.db`?

**Comment by mrvautin** [comment]
Weird. No all users are stored in the `users.db` file (or MongoDB).

Looking at the code. The logo path might be incorrect when using static files. Are you able to try changing Line 97 in `views/layouts/layout.hbs` to this:

`<a class="navbar-brand brand-image" href="{{app_context}}/"><img src="static/logo.png" width="140"></a>`

Unfortunately I don't have a Nginx setup I can test with.

**Comment by Kinzi** [comment]
Yeah, so the picture works with this link (just tested it in the browser - will update the files accordingly later).

For the users: so if I delete users.db I basically reset the whole thing and if I then sign up again it should be the admin user right?

**Comment by mrvautin** [comment]
Ok. Yeah if you delete the `users.db` it will show setup form again and user will be admin when setup. I'm thinking maybe it 404'd because of a route issue and caused user to not be created.

**Comment by Kinzi** [comment]
Ok deleted the db again and setup form shows. Would you mind trying to setup the admin user yourself, to see where the error is?
https://mydash.io/docs/setup

**Comment by mrvautin** [comment]
It's weird. The setup process works and a user is created as you cannot go to the `/setup` page after. Can you see if anything is being logged to the console when trying to login?

**Comment by Kinzi** [comment]
Don't have a timestamp unfortunately but i have a few of those in the pm2 errors:

```
Error: Not Found
    at /home/docs/app.js:241:15
    at Layer.handle [as handle_request] (/home/docs/node_modules/express/lib/router/layer.js:82:5)
    at trim_prefix (/home/docs/node_modules/express/lib/router/index.js:270:13)
    at /home/docs/node_modules/express/lib/router/index.js:237:9
    at Function.proto.process_params (/home/docs/node_modules/express/lib/router/index.js:312:12)
    at /home/docs/node_modules/express/lib/router/index.js:228:12
    at Function.match_layer (/home/docs/node_modules/express/lib/router/index.js:295:3)
    at next (/home/docs/node_modules/express/lib/router/index.js:189:10)
    at /home/docs/node_modules/express/lib/router/index.js:560:15
    at next (/home/docs/node_modules/express/lib/router/index.js:186:14)
```

**Comment by Kinzi** [comment]
Other than that it's just GET request. Should the signup log as a POST? If so it's not there...

**Comment by Kinzi** [comment]
Just checked the db files because I wanted to set the "is_admin":true manually.
But didn't work. Also I noticed, that when I update my user account it actually creates a new one....

What would be the right file permissions for the db file?

**Comment by Kinzi** [comment]
So I got it working now but for two problems:

1. In the frontend (based on the twitter theme) config is missing. I guess there is some file path wrong, but I can't find it. How is the config file loaded?
```
Uncaught ReferenceError: config is not defined
    at HTMLDocument.<anonymous> (openKB.js:8)
    at j (jquery.min.js:2)
    at k (jquery.min.js:2)
```

2. File upload seems to miss the app_context somewhere. Do you know where this path is set:
`POST https://mydash.io/file/upload_file 404 (Not Found)`

**Comment by mrvautin** [comment]
To be honest, I don't think many people have used the `app_context` setting. I will have a look. It's highly likely the inline upload doesn't honour the setting.

**Comment by mrvautin** [comment]
The DB files needs read/write permissions by the user running the app. I've pushed some fixes which should fix up both your outstanding issues and some others which you may not have discovered yet. Let me know how you go.

**Comment by Kinzi** [comment]
Hey this looks way better actually. :)

There is still one issue with the file upload, though. It uploads alright but the created url is still without `app_context` folder. 
Instead of:
`![file](/docs/uploads/inline_files/image.png)`
this tag is created:
`![file](/uploads/inline_files/image.png)`

Which means I have a broken image link in the editor and frontend(if I don't change it manually). :)

**Comment by Kinzi** [comment]
Also in twitter/layout.hbs at the end of `<head>` I had to alter to:

 ```
<script src="{{app_context}}/javascripts/jpushmenu.js"></script>
<script src="{{app_context}}/javascripts/openKB.js"></script>
{{#ifCond user_page '===' true}}
{{#if config.settings.google_analytics}}
{{{config.settings.google_analytics}}}
{{/if}}
{{/ifCond}}

```

and in views/layout.hbs:

```
{{#ifCond config.settings.show_website_logo '===' true}}
                    <a class="navbar-brand brand-image" href="{{app_context}}/"><img src="{{app_context}}/logo.png" width="140"></a>
                    {{else}}
```

Actually the whole `/static` thing became obsolete with the nginx setting above - it now works both ways. Having static might be more stable (or a fallback with a not-so-nice url).

**Comment by mrvautin** [comment]
Should be fixed in c4733e99fe68fb5b19a041eef6717ce29c28e9a8, 8945312c600dbab14cc9d9982b3af6d2a7c893a8 and cefc1c50fd8ae8c3e7b1e6147c1c2813aeefdad9.

**Comment by Kinzi** [comment]
@mrvautin Just to make this official: Thank you very much for your quick response and help. 👍  This library is awesome and I'm happy to have found it. Keep it up!

**Comment by mrvautin** [comment]
No worries. Glad you like it and can get some use from it.

**Comment by Kinzi** [comment]
Sorry I found one more:

The keywords/tags links at the bottom of an article miss the `app_context ` before /search:
`/app_context/search/myTag`

**Comment by mrvautin** [comment]
Fixed in 8195cb7aa65e2c4658615bcb2784c21e98aab1bc.
